blocks/extensions/__init__.py

```diff
@@ -363,7 +363,7 @@ def __init__(self, **kwargs):
         super(FinishAfter, self).__init__(**kwargs)
 
     def do(self, which_callback, *args):
-        self.main_loop.log.current_row.training_finish_requested = True
+        self.main_loop.log.current_row['training_finish_requested'] = True
```

Comment:
> If you write to the wrong key, you'll now get a KeyError when you try to read it

That does not always help because sometimes `get` is used to access the value. That is if one writes a custom extension that should finish training at some point and makes a typo, he will only notice it two days later when his training does not stop when it should have stopped. 

@dwf, what do you think?